### PR TITLE
FIP-15 TribalChief Combined PR

### PR DESCRIPTION
**Pull request by Joeysantoro**
PR including the DAO proposal steps for FIP-15.

Deployment: `deploy/fip_15.js`

- First create a TribalChief implementation (logic) contract
- Create a default OZ ProxyAdmin contract (imported in `contracts/external/TransparentUpgradeableProxy.sol`
- Transfer ownership of the admin to the DAO timelock
- Deploy a TransparentUpgradeableProxy for the TribalChief with the default admin. Atomically call initialize with the correct parameters
- Deploy Tribe dripper
- Deploy Optimistic Timelock

Proposal: `proposals/dao/fip_15.js` / `proposals/description/fip_15.json`
1. Withdraw TRIBE from old dripper
2. Allocate TRIBE to new dripper
3. Allocate first drip to TribalChief
4. Add FEI-TRIBE pair address
5. Add curve metapool address
6. Create TribalChief admin role
7. Grant TribalChief admin role to optimistic timelock
8. Send 7000 TRIBE to OA multisig
9. Revoke minter from FeiRewardsDistributorAddress

### CODE REVIEW

File: contracts/staking/StakingTokenWrapper.sol [hunk 1]
```diff
@@ -2,12 +2,7 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
-
-interface ITribalChief {
-    function stakedToken(uint256 _index) external view returns(IERC20);
-    function deposit(uint256 _pid, uint256 _amount, uint256 _lockLength) external;
-    function harvest(uint256 pid, address to) external;
-}
+import "./ITribalChief.sol";
```
Comment (sol-sr) [inline contracts/staking/StakingTokenWrapper.sol:5]: Do we want to just remove this completely?

File: test/staking/TribalChief.test.js [hunk 2]
```diff
@@ -213,7 +215,21 @@ describe('TribalChief', () => {
...
       this.coreRef = await MockCoreRef.new(this.core.address);
 
-      this.tribalChief = await TribalChief.new(this.core.address, this.tribe.address);
+      /// we can and probably should do this with a framework,
+      // but for now I'm going to write it out by hand.
+
+      // spin up the logic contract by hand
+      const tribalChief = await TribalChief.new(this.core.address);
+      // create a new proxy contract
+      const proxyContract = await Proxy.new(tribalChief.address, tribalChief.address, '0x', { from: userAddress });
```
Comment (sol-sr) [inline test/staking/TribalChief.test.js:224]: We should build the calldata to initialize things in this tx when creating this new proxy.

File: test/staking/TribalChief.test.js [hunk 3]
```diff
@@ -2031,7 +2199,21 @@ describe('TribalChief', () => {
...
       this.coreRef = await MockCoreRef.new(this.core.address);
 
-      this.tribalChief = await TribalChief.new(this.core.address, this.tribe.address);
+      /// we can and probably should do this with a framework,
+      // but for now I'm going to write it out by hand.
+
+      // spin up the logic contract by hand
+      const tribalChief = await TribalChief.new(this.core.address);
+      // create a new proxy contract
+      const proxyContract = await Proxy.new(tribalChief.address, tribalChief.address, '0x');
```
Comment (sol-sr) [inline test/staking/TribalChief.test.js:2208]: Same comment around initialization and creation, we can do this all in one step.

**Review by sol-sr — CHANGES_REQUESTED** [review]

File: contracts/staking/StakingTokenWrapper.sol (hunk 1 above)
Comment (Joeysantoro) [reply contracts/staking/StakingTokenWrapper.sol:5]: No we want it

File: test/staking/TribalChief.test.js (hunk 2 above)
Comment (Joeysantoro) [reply test/staking/TribalChief.test.js:224]: I think its fine as-is for the unit tests. We do the atomic calldata init on the main deploy and e2e tests

File: test/staking/TribalChief.test.js (hunk 3 above)
Comment (Joeysantoro) [reply test/staking/TribalChief.test.js:2208]: I think its fine as-is for the unit tests. We do the atomic calldata init on the main deploy and e2e tests

File: proposals/dao/fip_15.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    tribalChiefNumPools: (await tribalChief.numPools()).toString() === (2).toString(),
+    tribalChiefFEITRIBEUniswapPool: await tribalChief.stakedToken(0) === feiTribePairAddress,
+    tribalChiefCurve3FEIMetaPool: await tribalChief.stakedToken(1) === curve3MetapoolAddress,
+  };
+
+  expect(expectedValues.feiRewardsDistributorBalance).to.be.true;
+  expect(expectedValues.tribalChiefAllocPoints).to.be.true;
+  expect(expectedValues.tribalChiefNumPools).to.be.true;
+  expect(expectedValues.tribalChiefFEITRIBEUniswapPool).to.be.true;
+  expect(expectedValues.tribalChiefCurve3FEIMetaPool).to.be.true;
```
Comment (eswak) [inline proposals/dao/fip_15.js:139]: weird way to expect, you could have used : 
```js
expect(await tribe.balanceOf(feiRewardsDistributorAddress)).to.be.bignumber.equal('0');
...etc
```

but ok

**Pull request merged**